### DISCUSSION TITLE
[RESOLVED] How to parse Raw inbound message to hl7v2 message in js destination transformer ?

### DISCUSSION THREAD

**Discussion by fabrom** [nextgenhealthcare/connect#5131] [Q&A] [answered]
I havé to manage hl7 file with .ok file. 
I create a Channel with a raw data type source, a first destination to manage thé hl7 file and a second destination to manage the ok file.

 In the source section, i filtered destinations with destination filters on files extension.

In the hl7 destination, i now try to transform raw message from source to hl7 message in a first JavaScript Step.
And i havé à second step to change the hl7 message.

My first step don t work.

Please, What is the js code to transform raw to hl7 message that i can use in my second step ?
 
I use Mirth 3.12

**Comment by jonbartels** [comment, marked as answer]
I think you want `SerializerFactory.getSerializer` - Look in the "References" section for "Get Serializer". It shows a sample code block like this:

```
var dataType = 'HL7V2';
var serializationProperties = SerializerFactory.getDefaultSerializationProperties(dataType);
var deserializationProperties = SerializerFactory.getDefaultDeserializationProperties(dataType);
var serializer = SerializerFactory.getSerializer(dataType, serializationProperties, deserializationProperties);
```

That will deserialize any raw message to a structired HL7 message. 

A search on GitHub or the old forums might help show some examples - https://forums.mirthproject.io/search?q=getSerializer&searchJSON=%7B%22keywords%22%3A%22getSerializer%22%7D

**Comment by pacmano1** [comment]
Also if you cross post to here and the forums, please reference that in both places.   Question was answered over on the forums.

**Comment by fabrom** [comment]
Resolved by adding the following js code into my hl7 destination in the first transformer step 

```
msg = new XML(SerializerFactory.getSerializer('HL7V2').toXML(connectorMessage.getRawData()));
```
